### PR TITLE
ITN tests for BQMT

### PR DESCRIPTION
**Pull request by AnkitCLI**

### CODE REVIEW

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      List<JsonObject> bigQueryResponse2 = new ArrayList<>();
+      for (Object row : bigQueryRows2) {
+        JsonObject jsonData = gson.fromJson(String.valueOf(row), JsonObject.class);
+        bigQueryResponse2.add(jsonData);
+      }
+      boolean isValid = compareBigQueryDataAndBQMT(bigQueryResponse, bigQueryResponse2);
+      if (!isValid) {
+        return false; // Return false if validation fails for any table
+      }
+    return true; // Return true if validation passes for all tables
```
Comment (psainics) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java]: Bruh !

```java
return compareBigQueryDataAndBQMT(bigQueryResponse, bigQueryResponse2);
```

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        bigQueryResponse.add(jsonData);
+      }
+      List<Object> bigQueryRows2 = new ArrayList<>();
+      getBigQueryTableData(currentTargetTable, bigQueryRows2);
+
+      List<JsonObject> bigQueryResponse2 = new ArrayList<>();
+      for (Object row : bigQueryRows2) {
+        JsonObject jsonData = gson.fromJson(String.valueOf(row), JsonObject.class);
+        bigQueryResponse2.add(jsonData);
+      }
```
Comment (psainics) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java]: This logic can be extracted to a separate function to reduce code repetition !

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        tableNames.add(currentTableName);
+      }
+    }
+    if (tableNames.isEmpty()) {
+      throw new IllegalStateException("Tables not found."); // Throw an exception if no tables are found
+    }
+    return tableNames;
+  }
+  private static boolean compareBigQueryDataAndBQMT(List<JsonObject> bigQueryResponse, List<JsonObject> bqmtData) throws NullPointerException {
+    if (bigQueryResponse.size() != bqmtData.size()) {
```
Comment (psainics) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java]: NPE is an unchecked exception !
Is this required ?

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    return true; // Return true if validation passes for all tables
+  }
+
+  public static boolean validateBQToBigQueryMultiTableInOneTable(String sourceTable) throws IOException, InterruptedException {
+    String currentTargetTable = PluginPropertyUtils.pluginProp("bqTargetTable");
+   String currentSourceTable = sourceTable;
+      List<Object> bigQueryRows = new ArrayList<>();
+      getBigQueryTableData(currentSourceTable, bigQueryRows);
+      List<JsonObject> bigQueryResponse = new ArrayList<>();
+      for (Object row : bigQueryRows) {
```
Comment (psainics) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java]: Fix the indentation here !

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+import java.util.stream.Collectors;
+
+public class BigQueryMultiTableValidation {
+  static Gson gson = new Gson();
+  public static boolean validateBQToBigQueryMultiTableInTwoTables(List<String> sourceTables) throws IOException, InterruptedException {
+    List<String> targetTables = getTableByName().stream().sorted().collect(Collectors.toList());
+    if (sourceTables.size() != targetTables.size()) {
+      throw new IllegalArgumentException("Number of source tables and target tables must be the same.");
+    }
+    for (int i = 0; i < sourceTables.size(); i++) {
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java]: Either add comments for this validation logic or break it down.

File: src/e2e-test/resources/pluginParameters.properties [hunk 6]
```diff
@@ -1,6 +1,6 @@
 projectId=cdf-athena
 datasetprojectId=cdf-athena
-dataset=bq_automation
+dataset=ankitdataset
```
Comment (Vipinofficial11) [inline src/e2e-test/resources/pluginParameters.properties]: Change the name of the dataset, don't create a dataset with your name.

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java (hunk 1 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java]: done

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java (hunk 2 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java]: done

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java (hunk 3 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java]: yes this is required

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java (hunk 4 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java]: fixed.

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java (hunk 5 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java]: done.

File: src/e2e-test/resources/pluginParameters.properties (hunk 6 above)
Comment (AnkitCLI) [reply src/e2e-test/resources/pluginParameters.properties]: done.

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTable.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
+package io.cdap.plugin.bigquery.stepsdesign;
```
Comment (psainics) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTable.java:17]: License is missing here !

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java [hunk 8]
```diff
@@ -0,0 +1,117 @@
+package io.cdap.plugin.bigquery.stepsdesign;
```
Comment (psainics) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java:17]: License is missing here !

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTable.java (hunk 7 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTable.java:17]: added

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java (hunk 8 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java:17]: added

**Review by psainics — APPROVED** [review]
LGTM !

**Pull request merged**